wled00/src/dependencies/network/Network.cpp

```diff
@@ -3,12 +3,12 @@
 IPAddress NetworkClass::localIP()
 {
 #if defined(ARDUINO_ARCH_ESP32) && defined(WLED_USE_ETHERNET)
-  if (ETH.localIP()[0] != 0) {
-    return ETH.localIP();
-  }
+  IPAddress ipAddress = ETH.localIP();
```

Comment:
This is not equivalent. If `ETH.localIP()[0] == 0`, it should try `WiFi.localIP()` next. WiFi is still supported in ETH builds.

Other than that, a very welcome optimization :)